src/cbmc/cbmc_parse_options.cpp

```diff
@@ -1030,6 +1033,7 @@ void cbmc_parse_optionst::help()
     " --xml-ui                     use XML-formatted output\n"
     " --xml-interface              bi-directional XML interface\n"
     " --json-ui                    use JSON-formatted output\n"
+    " --json-interface             bi-directional JSON interface\n"
```

Comment:
Is this actually bi-directional?! Doesn't it only become "bi"directional when using both `--json-interface` and `--json-ui`? (The same is of course true for XML.)

Reply:
I was wondering about that and have now more investigated in the XML case (see comments there). The original intention was that --xml-interface subsumes --xml-ui. 
Actually, the current behaviour was broken because the UI was set before the option could have been parsed from stdin, which seems difficult to fix without splitting the `parse_optionst` constructor into two phases or adding a `set_ui` to `ui_message_handlert` (rather no).
Also, the question is what happens with errors during parsing the options - they should already be output in a proper format. Thus, I'd rather stick with the original intention of 'bi-directional',

Reply:
I'm not sure I fully understand - I agree that `--{json,xml}-interface` should subsume `--{json,xml}-ui` - but I don't think that was actually work as such? If things can be made to work as such then we'd have the desired bi-directional interface.

Reply:
`--json-interface` now implies `--json-ui`, and `--xml-interface` now implies `--xml-ui`.

>but I don't think that was actually work as such?

Not sure, I understand...

Reply:
Let me retry, my apologies for completely broken English:

1) A "bi"-directional interface is about input _and_ output.
2) `--{json,xml}-ui` set up the _output_ side of an interface.
3) The comment claims that `--{json,xml}-interface` configure a bi-directional interface.
4) The actual implementation, however, only reads _input_ when using `--{json,xml}-interface`.
5) One possible fix would be that upon seeing `--{json,xml}-interface` we also set the option for `--{json,xml}-ui`.
6) This might not actually work as it could be too late, in which case a more extensive change would be required. If this is the case, a simple fix would be to change the comment such as not to promise a bi-directional interface. Users wanting to use a bi-directional interface would then need to provide both `--{json,xml}-interface` _and_ `--{json,xml}-ui`.

Reply:
1. yes
2. yes
3. yes
4. yes
5. is being fixed here (was broken)
6. not required because 5 works.